core/src/main/scala-2/zio/config/TupleConversion.scala

```diff
@@ -0,0 +1,86 @@
+package zio.config
+
+import scala.language.experimental.macros
+import scala.reflect.macros.whitebox
+
+trait TupleConversion[A, B] {
+  def to(a: A): B
+  def from(b: B): A
+}
+object TupleConversion {
+  def apply[Prod, Tup]: TupleConversion[Prod, Tup] = macro genTupleConversion[Prod, Tup]
```

Comment:
I personally prefer just `A` and `B` over `Prod` and `Tup`.

Reply:
I prefer meaningful type parameter names. If it's the length, what about `P` and `T`? It is important that `Prod` must be the case class and `Tup` the tuple.

Reply:
`P` and `T` would be ok. Considering we used `K` and `V` for key and value across the app, and stays consistent with that approach.